ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/identity_provider_config_service.proto

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1.admin;
+
+import "google/protobuf/field_mask.proto";
+
+option java_outer_classname = "IdentityProviderConfigServiceOuterClass";
+option java_package = "com.daml.ledger.api.v1.admin";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1.Admin";
+
+// Service to manage identity provider configuration to make it possible for participant node administrators
+// to setup additional identity providers at runtime.
+
+// The fields of request messages (and sub-messages) are marked either as ``Optional`` or ``Required``:
+// (1) ``Optional`` denoting the client may leave the field unset when sending a request.
+// (2) ``Required`` denoting the client must set the field to a non-default value when sending a request.
+//
+// An identity provider config resource, once it has been created, can be modified.
+// In order to update the properties represented by the ``IdentityProviderConfig`` message use the ``UpdateIdentityProviderConfig`` RPC.
+// The only fields that can be modified are those marked as ``Modifiable``.
+service IdentityProviderConfigService {
+
+  // Create a new identity provider configuration.
+  // Errors:
+  // - ``ALREADY_EXISTS``: if the identity provider configuration already exists
+  // - ``UNAUTHENTICATED``: if the request does not include a valid access token
+  // - ``PERMISSION_DENIED``: if the claims in the token are insufficient to perform a given operation
+  // - ``INVALID_ARGUMENT``: if the payload is malformed or is missing required fields
```

Comment:
I think we don't need and shouldn't specify these error codes for new RPCs (e.g. we didn't specify there for the new `rpc UpdateUser` in `UserManagementService`). They should be superseded by error codes which are self-documenting.

Reply:
This is then confusing of having them in user management. What stops us to remove them from user management?

Reply:
Nothing :smile: would one of you @pbatko-da or @skisel-da be able to create the PR and shepherd it through?

Reply:
#15518